### PR TITLE
Added CSS2 color parsing: rgb(r, g, b)

### PR DESCRIPTION
**Pull request by jcward**
Per: https://www.w3.org/TR/SVG/types.html#DataTypeColor

Parses both numeric and % values, e.g.: rgb(255,127,0) == rgb(100%, 50%, 0)

Includes testcase.

**Side note:** We might need the test suite at some point to tolerate a slight difference in pixel color. Especially with opacity and % color -- a pixel is wholly counted as "incorrect" if it is even different by 1 in any R, G or B. This makes it easy to get false negatives by adjusting opacity / percentages in the testcases.

### CODE REVIEW

**Comment by ibilon** [comment]
On the side note: comparing rgb values is bad and not even correct (it's not linear), should use hsl or something similar.

File: src/format/svg/SVGData.hx [hunk 1]
```diff
@@ -723,5 +737,28 @@ class SVGData extends Group {
...
-	
+
+	private static inline function parseRGBMatch(rgbMatch):Int
+	{
+			// CSS2 rgb color definition, matches 0-255 or 0-100%
+			// e.g. rgb(255,127,0) == rgb(100%,50%,0)
+
+			inline function range(val:Float):Int {
+				// constrain to Int 0-255
+        if (val<0) val = 0;
```
Comment (ashes999) [inline src/format/svg/SVGData.hx]: Could you please use braces with `if` statements? It just makes the code slightly less error prone.

File: src/format/svg/SVGData.hx [hunk 2]
```diff
@@ -35,6 +35,7 @@ class SVGData extends Group {
 	private static var mScaleMatch = ~/scale\((.*)\)/;
 	private static var mMatrixMatch = ~/matrix\((.*)[, ](.*)[, ](.*)[, ](.*)[, ](.*)[, ](.*)\)/;
 	private static var mURLMatch = ~/url\(#(.*)\)/;
+	private static var mRGBMatch = ~/rgb\s*\(\s*(\d+)\s*(%)?\s*,\s*(\d+)\s*(%)?\s*,\s*(\d+)\s*(%)?\s*\)/;
```
Comment (ashes999) [inline src/format/svg/SVGData.hx]: I played around with this on [Rubular](http://rubular.com) and I see very strange results.

If I enter `rgb(255, 128, 0)`, I get six groups:
- 255
- Blank
- 128
- Blank
- 0
- Blank

I can also enter `rgb(255, 17%, 0)`, which is strange, but I'm not sure if it's worth the effort to throw an error if the user does something silly like this.

What do you think?

**Comment by jcward** [comment]
Yes, the 6 matches are intentional. Each of R, G, and B are parsed as `integer` and `optional_%`. If you don't use a percent, it's blank (or null in Haxe.) So `rgb(255,17%,0)` is supported. In CSS you can always write the 0 with or without any units, because 0 == 0% == 0px.

I've added braces.

**Comment by ashes999** [comment]
Looks okay to me.

@ibilon ?

**Comment by ibilon** [comment]
Looks good for me too.

**Pull request merged**